### PR TITLE
Add interactive graphic showing zdm migration phases in one presentat…

### PR DESCRIPTION
**Pull request by johnsmartco**
…ion with user navigation

### CODE REVIEW

**Comment by johnsmartco** [comment]
Good afternoon/morning @alicel, @mendonk, @emeliawilkinson24 , and @jgillenwater.  I've built and deployed the updated ZDM Introduction with the unified phases diagram. For UX and accessibility, the diagram includes user-controlled navigation and ARIA button labels, among other features.  It's also wicked cool.  See the temp output on coppi:

http://coppi-ssh.aws.dsinternal.org/en/test-graphic/docs/migrate/introduction.html

http://coppi-ssh.aws.dsinternal.org/en/test-graphic/docs/migrate/introduction.html#_interactive_walk_through_of_the_migration_phases

**Review by eric-schneider — CHANGES_REQUESTED** [review]
**tl;dr you need to enclose your alt text in double quotes `""`**

The square brackets `[]` at the end of an AsciiDoc macro (in this case, at the end of the `image::` macro) constitutes the macro's _attribute list_. As such, any content within the square brackets will be parsed as a list of attributes to be applied to the macro. This means that attribute [list parsing rules](https://docs.asciidoctor.org/asciidoc/latest/attributes/positional-and-named-attributes/#attribute-list-parsing) apply, and the use/order of certain special characters will affect how AsciiDoc interprets the content between the square brackets.

Furthermore, individual block/element types each have their own invisible mappings of [positional attributes](https://docs.asciidoctor.org/asciidoc/latest/attributes/positional-and-named-attributes/#positional). I won't go into detail, but the basic gist is that when you're writing out the attribute list syntax, you can define the attributes as either A.) a list of name-value pairs in any order; or B.) a list of values in a specific positional order. For example, the attribute list syntax we use for code blocks (`[source,javascript]`) utilizes positional attributes -- so long as the language is defined in the second attribute position (after the first comma) we don't have to use the name-value longhand syntax (e.g. `language=javascript`).

The `image::` macro has three positional attributes: 1 ⇒ alt (text), 2 ⇒ width, 3 ⇒ height. This means that the following two image macros are equivalent:

```
image::sunset.jpg[Sunset,300,400]
image::sunset.jpg[alt=Sunset,width=300,height=400]
```

However, alt text throws a wrench in the gears, because alt text often needs punctuation such as commas and periods, which are reserved characters for defining the attribute list itself. So when you're writing out your alt text, as soon as you place a comma, the AsciiDoc parser assumes the comma to be the end of the alt text string and expects the next thing after the comma to be a width value.

To prevent the parser from interpreting the punctuation marks within alt text, make sure to enclose your entire alt text string in double quotes (`""`). In general, I find it to be a best practice to always enclose alt text in double quotes. That way you know for sure that you won't cause attribute parsing issues, and it signals to other writers/contributors who may not know about the parsing rules that they too should enclose alt text in double quotes, thus preventing further issues in the future.

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 1]
```diff
@@ -47,46 +47,77 @@ For additional terms, see the xref:glossary.adoc[glossary].
 An important migration prerequisite is that you already have the matching schema on Target. A CQL statement that your client application sends to {zdm-proxy} must be able to succeed on both Origin and Target clusters. This means that any keyspace that your client application uses must exist on both Origin and Target with the same name. Table names must also match. For more, see xref:feasibility-checklists.adoc#_schemakeyspace_compatibility[Schema/keyspace compatibility].
 ====
 
-Your migration project occurs through a sequence of phases, which matches the structure of this ZDM documentation.
+Here's an overview of the migration phases, which matches the structure of this ZDM documentation.
 
 image::{imagesprefix}migration-all-phases.png[Migration phases from start to finish.]
```
Comment (jgillenwater) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc]: Why not replace this with the interactive graphic? It is redundant to have them in back-to-back sections.

**Review by jgillenwater — CHANGES_REQUESTED** [review]

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc (hunk 1 above)
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc]: Thanks @jgillenwater . Funny, @alicel  and I were just chatting about this very thing!  I proposed keeping both, to over-communicate what's about to happen in each phase, for users who are brand new to ZDM and easing into their understanding.  But if the redundancy occurred to you both, maybe I should remove the purple version.  Ok.  I'll comment out the first one.  Will need to tweak some text too.   :-)

Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc]: Done in most recent commit.  Will rebuild coppi version in a few minutes...

File: docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc [hunk 2]
```diff
@@ -1,3 +1,5 @@
+Now that you've seen a conceptual overview of the process, why not putting what you learned in practice?
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc:1]:
```suggestion
Now that you've seen a conceptual overview of the process, why not put what you learned into practice? 
```

Other options:
- "Now that you've seen a conceptual overview of the process, let's put what you learned into practice."
- "Now that you've seen a conceptual overview of the process, you can try putting what you learned into practice."

File: docs-src/zdm-core/modules/migrate/pages/change-read-routing.adoc [hunk 3]
```diff
@@ -7,7 +7,7 @@ This topic explains how you can configure the {zdm-proxy} to route all reads to
 
 Illustrated view of this phase:
 
-image::{imagesprefix}migration-phase4ra.png[Phase 4 diagram shows read routing on ZDM Proxy was switched to Target.]
+image::{imagesprefix}migration-phase4ra9.png[Phase 4 diagram shows read routing on ZDM Proxy was switched to Target.]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/change-read-routing.adoc:10]:
```suggestion
image::{imagesprefix}migration-phase4ra9.png["Phase 4 diagram shows read routing on ZDM Proxy was switched to Target."]
```

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 4]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
-Before your migration begins, you'll need to satisfy prerequisites, prepare your environment, and set up the recommended infrastructure.
+[.swiper]
+====
+[.slide]
+--
+.Walk-through the illustrated migration phases
+To get acquainted with the migration phases from start to finish, click the **Start** button below!
 
-=== Phase 1: Deploy ZDM Proxy and connect client applications
+image:migration-introduction9.png["Introductory page prompts you to click Start to begin the graphical presentation."]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc:61]:
```suggestion
image::migration-introduction9.png["Introductory page prompts you to click Start to begin the graphical presentation."]
```

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 5]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+--
 
-image::{imagesprefix}migration-phase2ra.png[Phase 2 diagram shows using tools to migrate data from Origin to Target.]
+[.slide] 
+--
+.Phase 2: Migrate data
+In this phase, we migrate existing data using Cassandra Data Migrator and/or DSBulk Migrator. Validate that the migrated data is correct, while continuing to perform dual writes.
 
-=== Phase 3: Async dual reads
+image:migration-phase2ra9.png["Migration Phase 2. We migrate existing data using Cassandra Data Migrator and/or DSBulk Migrator. Validate that the migrated data is correct, while continuing to perform dual writes."]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc:85]:
```suggestion
image::migration-phase2ra9.png["Migration Phase 2. We migrate existing data using Cassandra Data Migrator and/or DSBulk Migrator. Validate that the migrated data is correct, while continuing to perform dual writes."]
```

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 6]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
 
+[.slide] 
+--
+.Phase 3: Async dual reads
 In this phase, you can optionally enable asynchronous dual reads. The idea is to test performance and verify that Target can handle your application's live request load before cutting over from Origin to Target.
 
-image::{imagesprefix}migration-phase3ra.png[Phase 3 diagram shows optional step enabling async dual reads to test performance of Target.]
-
-=== Phase 4: Route reads to Target
+image:migration-phase3ra9.png["Migration Phase 3. Optionally enable asynchronous dual reads. Test performance and verify that Target can handle your application's live request load before cutting over from Origin to Target."]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc:93]:
```suggestion
image::migration-phase3ra9.png["Migration Phase 3. Optionally enable asynchronous dual reads. Test performance and verify that Target can handle your application's live request load before cutting over from Origin to Target."]
```

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 7]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+--
 
-In this phase, the read routing on the {zdm-proxy} is switched to Target so that all reads are executed on it, while writes are still sent to both clusters. In other words, Target becomes the primary cluster.
+[.slide] 
+--
+.Phase 4: Route reads to Target
+In this phase, the read routing on the ZDM Proxy is switched to Target so that all reads are executed on it, while writes are still sent to both clusters. In other words, Target becomes the primary cluster.
 
-image::{imagesprefix}migration-phase4ra.png[Phase 4 diagram shows read routing on ZDM Proxy was switched to Target.]
+image:migration-phase4ra9.png["Migration Phase 4. Read routing on ZDM Proxy is switched to Target so all reads are executed on it, while writes are still sent to both clusters. In other words, Target becomes the primary cluster."]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc:101]:
```suggestion
image::migration-phase4ra9.png["Migration Phase 4. Read routing on ZDM Proxy is switched to Target so all reads are executed on it, while writes are still sent to both clusters. In other words, Target becomes the primary cluster."]
```

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 8]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+--
 
-=== Phase 5: Connect directly to Target
+[.slide]
+-- 
+.Phase 5: Connect directly to Target
+In this phase, you'll move your client applications off the ZDM Proxy and connect the apps directly to Target. Once that happens, the migration is complete.
 
-In this phase, you'll move your client applications off the {zdm-proxy} and connect the apps directly to Target. Once that happens, the migration is complete.
+image:migration-phase5ra9.png["Migration Phase 5. Move your client applications off the ZDM Proxy and connect the apps directly to Target. Once that happens, the migration is complete."]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/pages/introduction.adoc:109]:
```suggestion
image::migration-phase5ra9.png["Migration Phase 5. Move your client applications off the ZDM Proxy and connect the apps directly to Target. Once that happens, the migration is complete."]
```

File: docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc [hunk 9]
```diff
@@ -1,3 +1,5 @@
+Now that you've seen a conceptual overview of the process, why not putting what you learned in practice? 
+
 We've built a complementary learning resource that is a companion to this comprehensive ZDM documentation. It's the {zdm-product} Interactive Lab, available for you here:
 
 link:https://www.datastax.com/dev/zdm[https://www.datastax.com/dev/zdm, window="_blank"]
```
Comment (eric-schneider) [inline docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc:5]:
```suggestion
https://www.datastax.com/dev/zdm[,window="_blank"]
```

**Review by eric-schneider — CHANGES_REQUESTED** [review]
I didn't call out each instance, but the phase descriptions don't seem to have a consistent voice. Some are active, and some are passive. I don't think that's a blocker, but it's something to consider for future improvement.

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 10]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
-
-image::{imagesprefix}pre-migration0ra.png[Diagram shows existing CQL-compatible environment before migration starts.]
-
-Before your migration begins, you'll need to satisfy prerequisites, prepare your environment, and set up the recommended infrastructure.
+[.swiper]
+====
+[.slide]
+--
+.Walk-through the illustrated migration phases
+To get acquainted with the migration phases from start to finish, click the **Start** button below!
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc:59]: Ah, no hyperbole allowed.  Ok.  :-)

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 11]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+--
 
-Let's look at Phase 1 of the migration. We'll deploy the ZDM Proxy instances and connect client applications to the proxies. This step activates the dual-write logic. Writes will be "bifurcated" (sent to both Origin and Target), while reads will be executed on Origin only.
+[.slide]
+--
+.Phase 0: Before your migration starts
+Let's look at a pre-migration, high-level view. At this point, your client applications are performing read/write operations with an existing CQL-compatible database. That is, Apache Cassandra®, DataStax Enterprise (DSE), or Astra DB.
 
-image::{imagesprefix}migration-phase1ra.png[Phase 1 diagram shows deployed ZDM Proxy instances, client app connections to proxies, and Target is setup.]
+image:pre-migration0ra9.png["A pre-migration high-level diagram. Your client applications perform read/write operations with an existing CQL-compatible database: Apache Cassandra®, DataStax Enterprise (DSE), or Astra DB."]
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc:69]: The image: single-colon syntax came from the S31 template, but I'll test it with image::

File: docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc (hunk 9 above)
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc:5]: I like the conversational "Now that you've..." line but will make a minor edit.  I'll also change the L5 link as you proposed (thanks).  That will come in the next commit. Resolving for now without this commit.

**Comment by johnsmartco** [comment]
> I didn't call out each instance, but the phase descriptions don't seem to have a consistent voice. Some are active, and some are passive. I don't think that's a blocker, but it's something to consider for future improvement.

Fixed in latest commit - thanks.

File: docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc (hunk 9 above)
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc]: Updated in latest commit.

**Comment by johnsmartco** [comment]
I think I've incorporated all the comments, except in cases noted in comments, and except for using single colon image: instead of image:: due to an unexpected result in output.  (Eric and I chatted separately about it.)

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 12]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+--
 
-=== Phase 2: Migrate data
+[.slide] 
+--
+.Phase 1: Deploy ZDM Proxy and connect client applications
+In this first phase, we'll deploy the ZDM Proxy instances and connect client applications to the proxies. This step activates the dual-write logic. Writes will be "bifurcated" (sent to both Origin and Target), while reads will be executed on Origin only.
 
-In this phase, we migrate existing data using {cstar-data-migrator} and/or {dsbulk-migrator}. Validate that the migrated data is correct, while continuing to perform dual writes.
+image:migration-phase1ra9.png["Migration Phase 1. We deploy ZDM Proxy instances and connect client applications to the proxies. This step activates the dual-write logic. Writes will be bifurcated (sent to both Origin and Target), while reads will be executed on Origin only."]
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc]: Done selectively in latest commit (no image::), pushed, and rebuilt on coppi. Thanks Eric.

File: docs-src/zdm-core/modules/migrate/pages/introduction.adoc [hunk 13]
```diff
@@ -47,46 +47,74 @@ For additional terms, see the xref:glossary.adoc[glossary].
...
+.Phase 2: Migrate data
+In this phase, we migrate existing data using Cassandra Data Migrator and/or DSBulk Migrator. Validate that the migrated data is correct, while continuing to perform dual writes.
 
-=== Phase 3: Async dual reads
+image:migration-phase2ra9.png["Migration Phase 2. We migrate existing data using Cassandra Data Migrator and/or DSBulk Migrator. Validate that the migrated data is correct, while continuing to perform dual writes."]
+--
 
+[.slide] 
+--
+.Phase 3: Async dual reads
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/introduction.adoc]: Done in latest commit, pushed, and rebuilt on coppi. Thanks Eric.

File: docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc (hunk 2 above)
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/partials/interactive-lab.adoc]: Done in latest commit, pushed, and rebuilt on coppi. Thanks Eric.

File: docs-src/zdm-core/modules/migrate/pages/change-read-routing.adoc (hunk 3 above)
Comment (johnsmartco) [reply docs-src/zdm-core/modules/migrate/pages/change-read-routing.adoc:10]: Quotes in place from an interim commit today (thanks). Didn't do image:: as discussed. Latest is on coppi.

**Review by eric-schneider — APPROVED** [review]
Approved. Nice work John!

**Review by jgillenwater — APPROVED** [review]
